spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java

```diff
@@ -96,6 +96,11 @@
 	 */
 	private boolean publisherReturns;
 
+	/**
+	 * Whether to enable simple publisher confirms.
+	 */
+	private boolean simplePublisherConfirms;
```

Comment:
@garyrussell could you please educate me on the difference between `publisherConfirms` and `simplePublisherConfirms`. Looking at those two properties, this feels a bit odd to me.

Perhaps we should refactor the configuration?

Reply:
@snicoll With standard publisher confirms, the caller gets a callback for each sent message, together with correlation data so he can figure out which sent message the confirmation is for.

Let's say we send 10 messages back-to-back; the broker might send us 10 discrete confirmations or one confirmation that says "the last 10 sends were good", depending on timing.

In the latter case, the framework still invokes the callback 10 times.

With simple confirms (added in spring-amqp 2.1), the caller is saying "I don't care to be told which messages succeeded or failed, I just want to wait until they all succeed"; the amap-client has a method `waitForConfirmsOrDie` which is called by the `RabbitTemplate` method with the same name.

See [Scoped Operations](https://docs.spring.io/spring-amqp/reference/html/#scoped-operations) in the reference guide.

You can still use simple confirms without it, but setting this property removes the additional scaffolding needed to support standard confirms.

Reply:
@garyrussell and I had a little chat offline and I shared having two properties for the same feature was a bit odd. Gary said he'd add an enum in Spring AMQP instead so we'll have to rework this PR to use that.

Wondering if switching a property's type in a feature release is acceptable so flagging for team attention.